js/api/Surveys.js

```diff
@@ -148,6 +169,21 @@ function clearSurveyCache(exclusions) {
     }
 
     realm.write(() => {
+      const expiredSurveysLength = expiredSurveys.length;
+      for (var i = 0; i < expiredSurveysLength; i++) {
+        let forms = realm.objects('Form').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+        let timeTriggers = realm.objects('TimeTrigger').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+        let geofenceTriggers = realm.objects('GeofenceTrigger').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+
+        let formsLength = forms.length;
+        for (var j = 0; j < formsLength; j++) {
+          let questions = realm.objects('Question').filtered(`formId= "${expiredSurveys[i].surveyId}"`)
```

Comment:
It seems that L180 it would be slightly cleaner to do a .forEach over forms.  Also couple places where let could be a const.